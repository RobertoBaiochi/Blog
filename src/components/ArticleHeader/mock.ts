import { ArticleHeaderProps } from '.';
import { data } from '../../api/data.json';

// const { posts } = data;
// const {
//   id,
//   attributes: { title, excerpt, createdAt },
// } = posts.data[1];

// export default {
//   id,
//   title,
//   excerpt,
//   cover: data.posts.data[1].attributes.cover,
//   author: data.posts.data[1].attributes.author.data,
//   categories: data.posts.data[1].attributes.categories.data,
//   createdAt,
// } as unknown as ArticleHeaderProps;

const {
  id,
  attributes: { title, excerpt, createdAt },
} = data.posts.data[1];

export default {
  id,
  title,
  excerpt,
  cover: data.posts.data[1].attributes.cover,
  author: data.posts.data[1].attributes.author.data,
  categories: data.posts.data[1].attributes.categories.data[1],
  createdAt,
} as unknown as ArticleHeaderProps;
