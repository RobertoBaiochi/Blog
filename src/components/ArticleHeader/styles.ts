import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacing.xlarge};
    margin-bottom: ${theme.spacing.xlarge};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};

    ${HeadingStyles} {
      margin: 0;
      margin-bottom: ${theme.spacing.small};
    }
  `}
`;

export const Excerpt = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacing.medium} 0;
    font-size: ${theme.font.sizes.medium};
  `};
`;

export const Cover = styled.img`
  ${({ theme }) => css`
    display: block;
    margin-bottom: ${theme.spacing.medium};
  `};
`;
