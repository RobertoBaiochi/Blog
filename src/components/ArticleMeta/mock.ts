import { ArticleMetaProps } from './';
import { data } from '../../api/data.json';

const { posts } = data;

const {
  attributes: {
    createdAt,
    author: {
      data: {
        id: idAuthor,
        attributes: { displayName: displayNameAuthor, slug: slugAuthor },
      },
    },
    categories,
  },
} = posts.data[1];

export default {
  createdAt,
  author: {
    idAuthor,
    displayNameAuthor,
    slugAuthor,
  },
  categories,
} as unknown as ArticleMetaProps;

// import { ArticleMetaProps } from '.';

// export default {
//   createdAt: '2023-05-09T04:49:38.694Z',
//   author: {
//     id: '1',
//     displayName: 'Roberto Baiochi',
//     slug: 'roberto-baiochi',
//   },
//   categories: [
//     {
//       id: '1',
//       displayName: 'ReactJs',
//       slug: 'react-js',
//     },
//     {
//       id: '2',
//       displayName: 'NextJs',
//       slug: 'next-js',
//     },
//   ],
// } as ArticleMetaProps;
