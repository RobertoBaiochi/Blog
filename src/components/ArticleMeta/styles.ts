import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${() => css`
    width: 100%;
  `}
`;
