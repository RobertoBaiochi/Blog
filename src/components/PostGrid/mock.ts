import { Posts } from '../../shared-typed/posts';
import { data } from '../../api/data.json';

export default { posts: data.posts.data } as unknown as Posts;
