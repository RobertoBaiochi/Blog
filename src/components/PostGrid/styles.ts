import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    width: 100%;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(32rem, 1fr));
    gap: ${theme.spacing.large};
  `};
`;
