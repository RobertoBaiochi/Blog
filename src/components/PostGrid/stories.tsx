import { Meta, StoryFn } from '@storybook/react';
import PostGrid, { PostGridProps } from '.';
import mock from './mock';

export default {
  title: 'PostGrid',
  component: PostGrid,
  args: mock,
} as Meta<PostGridProps>;

export const Template: StoryFn<PostGridProps> = (args) => {
  return (
    <div>
      <PostGrid {...args} />
    </div>
  );
};
