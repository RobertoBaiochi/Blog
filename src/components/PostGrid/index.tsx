import PostCard, { PostCardProps } from '../PostCard';
import { Wrapper, NotFound, GridContainer } from './styles';

export type PostGridProps = {
  posts?: PostCardProps[];
};

const PostGrid = ({ posts = [] }: PostGridProps) => {
  return (
    <Wrapper>
      {posts.length === 0 && <NotFound>Nenhum post encontrado aqui</NotFound>}

      <GridContainer>
        {posts.length > 0 &&
          posts.map((post) => {
            const {
              id,
              attributes: { slug },
            } = post;

            return <PostCard key={`${slug}+${id}`} {...post} />;
          })}
      </GridContainer>
    </Wrapper>
  );
};
export default PostGrid;
