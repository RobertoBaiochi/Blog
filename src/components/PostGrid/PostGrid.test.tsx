import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import PostGrid, { PostGridProps } from '.';
import '@testing-library/jest-dom';
import mock from './mock';

const props: PostGridProps = mock;
describe('<PostGrid />', () => {
  it('should render', () => {
    renderTheme(<PostGrid {...props} />);

    expect(screen.getByRole('heading', { name: 'Oi' })).toBeInTheDocument();
  });
});
