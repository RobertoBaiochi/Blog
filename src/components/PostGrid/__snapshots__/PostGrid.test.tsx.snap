// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostGrid /> should match snapshot 1`] = `
<div>
  <div
    class="sc-d5c95ca7-0 gDQVIO"
  >
    <div
      class="sc-d5c95ca7-2 eGbpfL"
    >
      <div
        class="sc-12652a66-0 epxRKc"
      >
        <a
          href="/post/como-criar-um-aplicativo-react-com-suporte-a-internacionalizacao-i18n"
        >
          <img
            alt="cidade na beira da praia com prédios ao final da tarde"
            class="sc-12652a66-1 iYcwyv"
            src="https://res.cloudinary.com/dfmbf9sae/image/upload/v1683607135/caio_silva_b1_BK_Bsi_JGI_unsplash_761a915801.jpg"
          />
        </a>
        <h2
          class="sc-b5aaf39e-0 kQHlLj"
        >
          <a
            href="/post/como-criar-um-aplicativo-react-com-suporte-a-internacionalizacao-i18n"
          >
            Como criar um aplicativo React com suporte a internacionalização (i18n)
          </a>
        </h2>
        <p
          class="sc-12652a66-2 jhiHKy"
        >
          Este artigo mostra como adicionar suporte a internacionalização em um aplicativo React usando as bibliotecas i18next e react-i18next. Inclui exemplos de tradução de texto e de como permitir que o usuário altere o idioma do aplicativo.
        </p>
      </div>
      <div
        class="sc-12652a66-0 epxRKc"
      >
        <a
          href="/post/usando-o-canvas-para-criar-graficos-em-java-script"
        >
          <img
            alt="praia vista de cima, água cor vende escuro e rochas"
            class="sc-12652a66-1 iYcwyv"
            src="https://res.cloudinary.com/dfmbf9sae/image/upload/v1683581086/adam_seckel_xhaz6_Nn_VT_64_unsplash_ad1c1db2e6.jpg"
          />
        </a>
        <h2
          class="sc-b5aaf39e-0 kQHlLj"
        >
          <a
            href="/post/usando-o-canvas-para-criar-graficos-em-java-script"
          >
            Usando o Canvas para criar gráficos em JavaScript
          </a>
        </h2>
        <p
          class="sc-12652a66-2 jhiHKy"
        >
          O artigo apresenta como usar a API Canvas em HTML e JavaScript para criar gráficos personalizados com estilização e interatividade, resultando em visualizações de dados impressionantes e envolventes.
        </p>
      </div>
      <div
        class="sc-12652a66-0 epxRKc"
      >
        <a
          href="/post/como-criar-um-aplicativo-de-podcast-com-next-js"
        >
          <img
            alt="campo com montanhas ao fundo em um dia nublado"
            class="sc-12652a66-1 iYcwyv"
            src="https://res.cloudinary.com/dfmbf9sae/image/upload/v1683608183/saira_G6jv_H_h_I5_Pc_unsplash_9938f8473d.jpg"
          />
        </a>
        <h2
          class="sc-b5aaf39e-0 kQHlLj"
        >
          <a
            href="/post/como-criar-um-aplicativo-de-podcast-com-next-js"
          >
            Como criar um aplicativo de podcast com Next.js
          </a>
        </h2>
        <p
          class="sc-12652a66-2 jhiHKy"
        >
          Este artigo ensina como criar um aplicativo de podcast utilizando o framework Next.js. Ele aborda a criação de componentes reutilizáveis, a realização de solicitações de API e a implantação do aplicativo na AWS.
        </p>
      </div>
    </div>
  </div>
</div>
`;
